### ISSUE TITLE
Passive Logging overkill

### ISSUE THREAD

**Issue by Failure404** [mendhak/gpslogger#766]
Hi,
using GPS Logger for quite a while.
Recent changes to passive location logging made the app unusable for me.
I am posting locations using custom URL to my server. Since I can't set any filters, locations get posted EVERY second. That is total overkill for a mobile connection. Massive battery drain, and GPS logger frequently crashes when there is a problematic mobile connection.
Is it not possible to
Check every x seconds if a passive location is available and
only upload it in a specified interval?
I really only want passive locations being logged, and not all of them - only the one which is available at a certain interval. If none is available, wait for next interval.
Battery drain is so massive that it makes passive location logging pointless. Using GPS full time would probably use less battery.

**Comment by mendhak** [comment]
Yes, the passive location started ignoring the time filter in v100, that was done in #705 and the idea here is that the phone has already spent the energy getting the location, so the app can read it and log it.

While it's probably possible to make it take on the time filter too, I'm not keen on making the passive option a dual-option listener (ie : passive without filters, passive with filters) as - the unfortunate bit -  the code is already quite complex so it's better to pick just one way.

**Comment by mendhak** [comment]
You had mentioned, when there's a problematic mobile connection you're getting crashes?  Is it crashes with an ANR dialog??  Are you able to send a report.  Or [send a debug log](https://gpslogger.app/#troubleshooting) when that happens?  Right now the URL logger should try a few times and handle exceptions before giving up.  The sending occurs in a background thread so you shouldn't be seeing crashes in the main app.

**Comment by Failure404** [comment]
Thank you kindly for your reply.
I just leave the app running in the background posting passive locations to a custom url, no logging to a file or similar. Once in a while I get a "app has crashed" popup while it's running in the background. It happens rarely. My thought was that it must have to do with the fact that it tries every second to post data to a custom url, which sometimes can be problematic with mobile connections switching between cells and wifis. Will try to get a debug log / send report.
Regarding battery consumption, I think it needs alot of battery when sending data to a custom url every second...
It would be nice if I could just leave it running in the background, grab a passive location every 5 minutes and post that to a custom url if it has changed.
Thank you for your work.

**Comment by mendhak** [comment]
The battery consumption problem is likely two-fold.  If you are getting passive locations every second, then there is something else that is causing locations to be read so frequently.  Some other service or application is reading locations frequently, and if that is unintentional that's a good place to start investigating. 

Of course as you've seen with the passive locations triggering a custom-url send.  Note that passive location reads are meant to be arbitrary, adhoc, reactive, rather than periodic.  If you really want a custom URL send every _x_ minutes then it's better to ignore the passive location provider entirely and just use normal network/satellite locations with a time interval of 5 minutes.

**Comment by Terminatorthre** [comment]
Hi,
sorry to comment to this very old issue but I guess it is still somewhat open. 
I have a situation where I want to get many location updates from a specific device via custom URL (Nextcloud-Phonetrack) when it is moving. This is working fine with passive locations. 
The problem that I am seeing is that the connection does not allow sending the data fast enough. I can transfer about 1 or 2 locations per second through 4G network but get about 3 or 4 passive location updates per second. This leads to a situation where the server falls behind the current location of the device even though it is online.
My ideas would be:
1. Limit the amount of passive location updates e.g. to 1/s or configurable
2. Make the requests to the custom URL faster (maybe multiple threads?) but this also depends on the server capabilites
3. Send the newest element from the queue first instead of the oldest. This way the server has the most up-to-date location of the device and can receive history later.

Any thoughs on that?
